### PR TITLE
[bitnami/redis] Added HPA for redis

### LINKED ISSUE #6551: [bitnami/magento], [bitnami/redis] Add HorizontalPodAutoscaler to chart
Edit: added Redis to feature suggestion as well.

**Which chart**:
bitnami/magento, bitnami/redis, latest

**Is your feature request related to a problem? Please describe.**
It'd be good to be able to enable a HPA for Magento + Elasticsearch, and Redis and not having to create and manage one manually.

**Describe the solution you'd like**
Being able to enable autoscaling and set settings via helm chart values would be good. Something like https://github.com/kubernetes/ingress-nginx/blob/9e89951ec76045f69058a6bd68692715364627d5/charts/ingress-nginx/values.yaml#L318

**Describe alternatives you've considered**
Creating manually, but doing this means it is managed separately which is not ideal.

### PR DESCRIPTION
**Pull request by mahgo**
**Description of the change**

Added Horizontal Pod Autoscaler for the Redis chart

**Benefits**
Being able to enable a HPA when using the chart.

**Applicable issues**

  - fixes #6551

**Additional information**
I was not able to test this out - for some reason the pods were already getting into a CrashLoopBackOff before I even changed anything, is anyone else having this problem?

**Checklist** 

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

File: bitnami/redis/values.yaml [hunk 1]
```diff
@@ -1000,6 +1008,14 @@ pdb:
   ## @param pdb.maxUnavailable Max number of pods that can be unavailable after the eviction
   ##
   maxUnavailable:
+## Autoscaling configuration
```
Comment (alvneiayu) [inline bitnami/redis/values.yaml:1011]: Could you move all of those values inside the replica section, please?. I mean, the autoscaling is going to work only for the replicas and right now the values are in a section shared with the master and the replicas.

**Review by alvneiayu — CHANGES_REQUESTED** [review]
Hi @mahgo

I reviewed the patch. Everything looks good to me but I think that it is better to move the autoscaling values inside the replica section.

Again, thanks a lot for your time and your work.

Álvaro

File: bitnami/redis/values.yaml (hunk 1 above)
Comment (mahgo) [reply bitnami/redis/values.yaml]: Done

File: bitnami/redis/templates/replicas/hpa.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.autoscaling.enabled }}
+{{- if .Values.replicate.autoscaling.enabled }}
```
Comment (alvneiayu) [inline bitnami/redis/templates/replicas/hpa.yaml:1]: {{- if .Values.replica.autoscaling.enabled }}

**Review by alvneiayu — COMMENTED** [review]
hi @mahgo 

I saw that you are having some linter errors. You have a Typo using "replicate" instead of "replica".

Thanks a lot for your time.

Álvaro

File: bitnami/redis/templates/replicas/hpa.yaml (hunk 2 above)
Comment (mahgo) [reply bitnami/redis/templates/replicas/hpa.yaml]: Thanks, fixed.

**Review by alvneiayu — APPROVED** [review]
LGTM

**Comment by kalali** [comment]
If sentinel is used the SS name is derived using ` {{ printf "%s-node" (include "common.names.fullname" .) }}` while when there is no sentinel the SS name is derived using ` {{ printf "%s-replicas" (include "common.names.fullname" .) }}`. The hpa needs to take that into account.

**Comment by alvneiayu** [comment]
hi @kalali 

Thanks a lot for notifying this problem. We are open to receive PRs and we will really happy to review it.

Thanks a lot again for your time and your help

Álvaro

**Pull request merged**